onadata/libs/utils/csv_import.py

```diff
@@ -251,9 +251,43 @@ def submit_csv(username, xform, csv_file, overwrite=False):
     xl_datetime_columns = [
         dt.get('name') for dt in x_json.get('children')
         if dt.get('type') in XLS_DATETIME_FIELDS]
+    xl_integer_columns = [
+        dt.get('name') for dt in x_json.get('children')
+        if dt.get('type') == 'integer'
+    ]
+    xl_decimal_columns = [
+        dt.get('name') for dt in x_json.get('children')
+        if dt.get('type') == 'decimal'
+    ]
 
     try:
         for row in csv_reader:
+            for key in xl_integer_columns:
+                # For some reason Integer fields are not validated
+                # We check that the passed in data is a valid interger
+                val = row.get(key, '')
+                if val:
+                    try:
+                        interger = int(val)
+                    except ValueError:
+                        raise Exception(
+                                'Unknown interger format: {}'.format(val))
```

Comment:
spelling

Reply:
Ah didn't notice that, I'll change this and the rest to the proper spelling.

Reply:
Fixed in latest commits